File: telegram-bot-core/src/main/java/ru/tggc/telegrambotcore/dto/UserDto.java
package ru.tggc.telegrambotcore.dto;

import java.util.Objects;

public record UserDto(
        Long userId,
        String username
) {
    @Override
    public boolean equals(Object o) {
        if (o == null || getClass() != o.getClass()) return false;
        UserDto userDto = (UserDto) o;
        return Objects.equals(userId, userDto.userId);
    }

    @Override
    public int hashCode() {
        return Objects.hashCode(userId);
    }
}

File: telegram-bot-core/src/main/kotlin/ru/tggc/telegrambotcore/dto/UserDto.kt
package ru.tggc.telegrambotcore.dto

import java.util.*

@JvmRecord
data class UserDto(
    val userId: Long,
    val username: String
) {
    override fun equals(other: Any?): Boolean {
        if (other == null || javaClass != other.javaClass) return false
        val userDto = other as UserDto
        return userId == userDto.userId
    }

    override fun hashCode(): Int {
        return Objects.hashCode(userId)
    }
}
